CMakeLists.txt

```
@@ -23,6 +23,8 @@ set(plugin_SRC src/RobotModelUpdate.cpp)
 
 set(plugin_HDR src/RobotModelUpdate.h)
 
+option(AUTOLOAD_RobotModelUpdate_PLUGIN
+       "Automatically load ${PLUGIN_NAME} plugin" OFF)
```

I think we should use this instead:
```suggestion
set(AUTOLOAD_RobotModelUpdate_PLUGIN
    OFF
    CACHE INTERNAL "Automatically load ${PLUGIN_NAME} plugin"
)
```

The `add_plugin` macro is the one adding the option in the first place, and what you want to do is set the default value to `OFF`. I think redeclaring the option as you did has no effect (some existing plugins do that but it doesn't actually work)